tests/playbooks/roles/install-csi-cinder/tasks/main.yaml

```diff
@@ -0,0 +1,143 @@
+---
+- name: Ensure CPO source folder
+  shell:
+    executable: /bin/bash
+    cmd: |
+      rm -rf $GOPATH/src/k8s.io/cloud-provider-openstack
+      mkdir -p $GOPATH/src/k8s.io; cd $_
+      git clone https://github.com/kubernetes/cloud-provider-openstack
+      cd cloud-provider-openstack
+      git fetch origin +refs/pull/{{ github_pr }}/merge
+      git checkout FETCH_HEAD; git checkout -b PR{{ github_pr }}
+
+- name: Build and upload cinder-csi-plugin image
+  shell:
+    executable: /bin/bash
+    cmd: |
+      cd $GOPATH/src/k8s.io/cloud-provider-openstack
+
+      REGISTRY={{ image_registry_host }} \
+      VERSION={{ github_pr }} \
+      IMAGE_NAMES=cinder-csi-plugin \
+      make upload-image-amd64
+
+- name: Prepare cloud config
+  shell:
+    executable: /bin/bash
+    cmd: |
+      # kubectl -n kube-system get secret cloud-config >/dev/null 2>&1
+      #if [ $? -eq 0 ]; then
+      #  exit 0
+      #fi
+
+      set -ex
+
+      set +x; source {{ devstack_workdir }}/openrc admin admin > /dev/null; set -x
+      tenant_id=$(openstack project show demo -c id -f value)
+      sudnet_id=$(openstack subnet show private-subnet -c id -f value)
+      external_network_id=$(openstack network list --external -c ID -f value)
+
+      set +x; source {{ devstack_workdir }}/openrc demo demo > /dev/null; set -x
+      cat <<EOF > {{ ansible_user_dir }}/cloud.conf
+      [Global]
+      auth-url=${OS_AUTH_URL}/v3
+      username=${OS_USERNAME}
+      password=${OS_PASSWORD}
+      region=${OS_REGION_NAME}
+      tenant-id=$tenant_id
+      domain-id=default
+
+      [LoadBalancer]
+      use-octavia=true
+      subnet-id=$sudnet_id
+      floating-network-id=$external_network_id
+
+      [BlockStorage]
+      bs-version=v2
+      EOF
+
+      kubectl create secret -n kube-system generic cloud-config --from-file={{ ansible_user_dir }}/cloud.conf
+
+      # replace manifest cloud secret file
+      b64data=`cat {{ ansible_user_dir }}/cloud.conf | base64 -w 0`
+      cd $GOPATH/src/k8s.io/cloud-provider-openstack
+
+      sed -i "/cloud\.conf/c\  cloud.conf: $b64data" manifests/cinder-csi-plugin/csi-secret-cinderplugin.yaml
+
+- name: Deploy cinder-csi-plugin
+  shell:
+    executable: /bin/bash
+    cmd: |
+      cd $GOPATH/src/k8s.io/cloud-provider-openstack
+      kubectl apply -f manifests/cinder-csi-plugin
+
+- name: Wait for csi-cinder-controllerplugin up and running
+  shell:
+    executable: /bin/bash
+    cmd: |
+      kubectl -n kube-system get pod | grep csi-cinder-controllerplugin  | grep Running
+  register: check_csi
+  until: check_csi.rc == 0
+  retries: 24
+  delay: 5
+  ignore_errors: yes
+
+- name: Wait for csi-cinder-nodeplugin up and running
+  shell:
+    executable: /bin/bash
+    cmd: |
+      kubectl -n kube-system get pod | grep csi-cinder-nodeplugin  | grep Running
+  register: check_csi
+  until: check_csi.rc == 0
+  retries: 24
+  delay: 5
+  ignore_errors: yes
+
+- name: create volume snapshot CRD and controllers
```

Comment:
please check if snapshot controller is running here as well

Reply:
I will remove those snapshot controller apply

we are having snapshot issue now, so should be ok to add it later on if needed
I doubt because all other csi test should have this snapshot so does e2e framework alredy have it 
is a question mark to me , so need more check